py/objdict.c

```diff
@@ -105,6 +105,12 @@ STATIC mp_obj_t dict_unary_op(mp_uint_t op, mp_obj_t self_in) {
     switch (op) {
         case MP_UNARY_OP_BOOL: return mp_obj_new_bool(self->map.used != 0);
         case MP_UNARY_OP_LEN: return MP_OBJ_NEW_SMALL_INT(self->map.used);
+        #if MICROPY_PY_SYS_GETSIZEOF
+        case MP_UNARY_OP_SIZEOF: {
+            mp_int_t sz = sizeof(*self) + sizeof(*self->map.table) * self->map.alloc;
```

Comment:
Fixed. This is just another illustration of concerns expressed in https://github.com/micropython/micropython/pull/3111 - some people didn't master even size_t usage, and there're ideas to introduce even more obscure stuff ;-).